### PR TITLE
Fixes the blind spell.

### LINKED ISSUE #4745: blind spell broken
1. try using blind spell on someone

2. fails. error message is You shouldn't have this spell! Something's wrong.

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

- Rewrites the blind spell to not be ancient
- Snaps the inflict_handler file since it's unused
- Groups the spell blindness traits into one (MAGIC_BLIND)
- Fixes minor error in the spelling of the book granter
- Some general spell optimisations

## Why It's Good For The Game

Blindness works again, and is cleaner.

Closes #4745

## Changelog
:cl:
del: old "inflict_handler" file
fix: blind spell works again
refactor: refactors the blind spell to not be ancient
/:cl:

### CODE REVIEW

File: code/modules/spells/spell_types/blind.dm [hunk 1]
```diff
@@ -1,34 +1,47 @@
...
-	amt_eye_blurry = 20
-	sound = 'sound/magic/blind.ogg'
 
-/obj/effect/proc_holder/spell/targeted/genetic/blind
-	mutations = list(BLINDMUT)
-	duration = 300
-	charge_max = 400 // needs to be higher than the duration or it'll be permanent
-	sound = 'sound/magic/blind.ogg'
+/obj/effect/proc_holder/spell/targeted/blind/cast(list/targets, mob/user = usr)
+	if(!targets.len)
```
Comment (utilbird) [inline code/modules/spells/spell_types/blind.dm]: `length()` is more performant than `.len` if you're just checking list length :)
```suggestion
	if(!length(targets))
```

File: code/modules/spells/spell_types/blind.dm [hunk 2]
```diff
@@ -1,34 +1,47 @@
...
+	var/mob/living/carbon/target = targets[1]
 
-/obj/effect/proc_holder/spell/pointed/trigger/blind/intercept_check(mob/user, atom/target)
-	if(!..())
-		return FALSE
-	if(!isliving(target))
-		to_chat(user, "<span class='warning'>You can only blind living beings!</span>")
-		return FALSE
-	return TRUE
+	if(!is_type_in_typecache(target, compatible_mobs_typecache))
```
Comment (utilbird) [inline code/modules/spells/spell_types/blind.dm]:
```suggestion
	if(!compatible_mobs_typecache[target.type])
```

File: code/modules/spells/spell_types/blind.dm [hunk 3]
```diff
@@ -1,34 +1,47 @@
...
+		return
+	
+	target.visible_message("<span class='danger'>[target]'s eyes darken as black smoke starts coming out of them!</span>", \
+						   "<span class='danger'>Your eyes hurt as they start smoking, you panic as you realise you're blind!</span>")
+	target.emote("scream")
+	target.become_blind(MAGIC_BLIND)
+	addtimer(CALLBACK(src, .proc/cure_blindness, target), duration)
+
+/obj/effect/proc_holder/spell/targeted/blind/proc/cure_blindness(mob/target)
+	var/mob/living/L = target
```
Comment (utilbird) [inline code/modules/spells/spell_types/blind.dm]:
```suggestion
/obj/effect/proc_holder/spell/targeted/blind/proc/cure_blindness(mob/living/L)
```

File: code/modules/spells/spell_types/blind.dm (hunk 2 above)
Comment (PowerfulBacon) [reply code/modules/spells/spell_types/blind.dm]: Since this is in the cast rather than the check will it not still use the cooldown if it fails?

Comment (ivanmixo) [reply code/modules/spells/spell_types/blind.dm]: Yeah, this still triggers the cooldown. Seems fine tbh.

Comment (PowerfulBacon) [reply code/modules/spells/spell_types/blind.dm]: The cooldown shouldnt run if you click on nothing or a mob you can't blind

File: code/modules/spells/spell_types/blind.dm [hunk 4]
```diff
@@ -1,34 +1,50 @@
...
+		to_chat(user, "<span class='notice'>[target.p_theyre(TRUE)] too far away!</span>")
+		revert_cast()
+		return
+	
+	if(target.anti_magic_check() || HAS_TRAIT(target, TRAIT_WARDED))
+		to_chat(user, "<span class='warning'>The spell had no effect!</span>")
+		target.visible_message("<span class='danger'>[target]'s eyes darken, but instantly turn back to their regular color, leaving [target] unharmed!</span>", \
+						   "<span class='danger'>Your eyes hurt for a moment, but the blindness is repulsed by your anti-magic protection!</span>")
+		revert_cast()
+		return
```
Comment (PowerfulBacon) [inline code/modules/spells/spell_types/blind.dm]: This shouldn't revert the cast

**Review by PowerfulBacon — APPROVED** [review]

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**